### PR TITLE
add io.github.mahmoudbahaa.outlook_for_linux

### PR DESCRIPTION
**Pull request by mahmoudbahaa**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

**Comment by mahmoudbahaa** [comment]
bot, build com.github.mahmoudbahaa.outlook_for_linux

**Comment by mahmoudbahaa** [comment]
bot, build io.github.mahmoudbahaa.outlook_for_linux

**Comment by mahmoudbahaa** [comment]
bot, build io.github.mahmoudbahaa.outlook_for_linux

File: flathub.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+    "only-arches": ["x86_64"]
```
Comment (bbhtt) [inline flathub.json:2]: Is aarch64 available? Can you build from source?

File: io.github.mahmoudbahaa.outlook_for_linux.appdata.xml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+			<description>
+				<p> Initial release same as teams-for-linux 1.3.13 but for outlook</p>
+				<p> See https://github.com/IsmaelMartinez/teams-for-linux for actual changelog </p>
+			</description>
+		</release>
+	</releases>
+	<developer_name>Mahmoud Bahaa</developer_name>
+	<screenshots>
+		<screenshot type="default">
+			<image type="source">https://iili.io/sUyAc7.png</image>
```
Comment (bbhtt) [inline io.github.mahmoudbahaa.outlook_for_linux.appdata.xml:28]: Host this somewhere else, more reliable like in your upstream repo

File: io.github.mahmoudbahaa.outlook_for_linux.appdata.xml
Comment (bbhtt) [inline io.github.mahmoudbahaa.outlook_for_linux.appdata.xml:1]: This file should be in upstream repo.

File: io.github.mahmoudbahaa.outlook_for_linux.appdata.xml [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+		</screenshot>
+	</screenshots>
+	<content_rating type="oars-1.0">
```
Comment (bbhtt) [inline io.github.mahmoudbahaa.outlook_for_linux.appdata.xml:31]: Use 1.1 in the generator then you can ommit the none list

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 4]
```diff
@@ -0,0 +1,57 @@
+app-id: io.github.mahmoudbahaa.outlook_for_linux
+base: org.electronjs.Electron2.BaseApp
+base-version: '22.08'
```
Comment (bbhtt) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:3]:
```suggestion
base-version: '23.08'
```

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '22.08'
```
Comment (bbhtt) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:5]:
```suggestion
runtime-version: '23.08'
```

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=all
+  - --filesystem=home
+  - --filesystem=xdg-run/dconf
+  - --filesystem=~/.config/dconf:ro
+  - --filesystem=xdg-run/pipewire-0
+  - --env=DCONF_USER_CONFIG_DIR=.config/dconf
+  - --talk-name=ca.desrt.dconf
```
Comment (bbhtt) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:21]: Why are these needed?

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=org.freedesktop.Notifications
```
Comment (bbhtt) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:23]:
```suggestion
```

Not required with 23.08

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  - --talk-name=org.freedesktop.PowerManagement
+  - --talk-name=org.freedesktop.ScreenSaver
+  - --talk-name=org.gtk.Notifications
+  - --talk-name=org.gnome.SessionManager
+  - --env=XDG_CURRENT_DESKTOP=Unity
```
Comment (bbhtt) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:28]: Why?

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+  - --filesystem=xdg-run/dconf
+  - --filesystem=~/.config/dconf:ro
+  - --filesystem=xdg-run/pipewire-0
+  - --env=DCONF_USER_CONFIG_DIR=.config/dconf
+  - --talk-name=ca.desrt.dconf
+  - --talk-name=org.kde.StatusNotifierWatcher
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.freedesktop.PowerManagement
+  - --talk-name=org.freedesktop.ScreenSaver
+  - --talk-name=org.gtk.Notifications
```
Comment (bbhtt) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:26]: Is this actually used? Apps are supposed to use the FDO notifications

File: io.github.mahmoudbahaa.outlook_for_linux.appdata.xml [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+	</description>
+	<url type="homepage">https://github.com/mahmoudbahaa/outlook-for-linux</url>
+	<url type="help">https://github.com/mahmoudbahaa/outlook-for-linux/blob/develop-outlook/KNOWN_ISSUES.md</url>
+	<url type="bugtracker">https://github.com/mahmoudbahaa/outlook-for-linux/issues</url>
+	<launchable type="desktop-id">io.github.mahmoudbahaa.outlook_for_linux.desktop</launchable>
+	<releases>
+		<release version="1.3.13" date="2023-10-17">
+			<description>
+				<p> Initial release same as teams-for-linux 1.3.13 but for outlook</p>
+				<p> See https://github.com/IsmaelMartinez/teams-for-linux for actual changelog </p>
```
Comment (bbhtt) [inline io.github.mahmoudbahaa.outlook_for_linux.appdata.xml:21]: This will fail validation, use url type details tag

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+command: outlook-for-linux
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=wayland
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=all
+  - --filesystem=home
```
Comment (bbhtt) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:16]: Why do you need full home? Does it support filechooser portals?

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+sdk: org.freedesktop.Sdk
+command: outlook-for-linux
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=wayland
+  - --socket=x11
+  - --socket=pulseaudio
+  - --share=network
+  - --device=all
```
Comment (bbhtt) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:15]: Seems like all you would need is dri?

**Comment by mahmoudbahaa** [comment]
bot, build io.github.mahmoudbahaa.outlook_for_linux

File: io.github.mahmoudbahaa.outlook_for_linux.appdata.xml (hunk 3 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.appdata.xml]: ok done

File: io.github.mahmoudbahaa.outlook_for_linux.appdata.xml (hunk 2 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.appdata.xml]: done

File: io.github.mahmoudbahaa.outlook_for_linux.yml (hunk 4 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.yml]: done

File: io.github.mahmoudbahaa.outlook_for_linux.yml (hunk 5 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.yml]: done

File: io.github.mahmoudbahaa.outlook_for_linux.yml (hunk 7 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.yml]: done

**Comment by mahmoudbahaa** [comment]
bot, build io.github.mahmoudbahaa.outlook_for_linux

File: io.github.mahmoudbahaa.outlook_for_linux.appdata.xml (hunk 10 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.appdata.xml]: done

**Comment by mahmoudbahaa** [comment]
bot, build io.github.mahmoudbahaa.outlook_for_linux

File: io.github.mahmoudbahaa.outlook_for_linux.yml (hunk 12 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.yml]: This was needed for teams but you are right seems it isn't needed for outlook removed

**Comment by mahmoudbahaa** [comment]
bot, build io.github.mahmoudbahaa.outlook_for_linux

**Comment by mahmoudbahaa** [comment]
bot, build io.github.mahmoudbahaa.outlook_for_linux

**Comment by mahmoudbahaa** [comment]
bot, build io.github.mahmoudbahaa.outlook_for_linux

**Comment by mahmoudbahaa** [comment]
bot, build io.github.mahmoudbahaa.outlook_for_linux

**Comment by mahmoudbahaa** [comment]
bot, build io.github.mahmoudbahaa.outlook_for_linux

**Comment by bbhtt** [comment]
Can you please trigger a build after doing all of the changes instead after each push? Build it locally to test.

**Comment by mahmoudbahaa** [comment]
yeah sure can you share how to build it locally though

**Comment by bbhtt** [comment]
Install flatpak-builder from your distribution or this flatpak package https://github.com/flathub/org.flatpak.Builder

Then `flatpak-builder build --force-clean <manifest>` or `flatpak run org.flatpak.Builder build --force-clean <manifest>

To install it locally: `flatpak-builder build --force-clean --user --install <manifest>`, then run it with `flatpak run <app-id>`

You'll need to install the following from flathub before starting the build:

```
flatpak install org.electronjs.Electron2.BaseApp//23.08
flatpak install org.freedesktop.Sdk//23.08
flatpak install org.freedesktop.Platform//23.08
```

File: flathub.json (hunk 1 above)
Comment (hfiguiere) [reply flathub.json:2]: this question hasn't been answered.

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+base-version: '23.08'
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: outlook-for-linux
+separate-locales: false
+finish-args:
+  - --share=ipc
+  - --socket=wayland
+  - --socket=x11
```
Comment (hfiguiere) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:12]: does this really support wayland? If not remove like 11, otherwise line 12 should be `fallback-x11`

(I'm pretty it's the former)

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 14]
```diff
@@ -0,0 +1,50 @@
...
+  - --socket=pulseaudio
+  - --share=network
+  - --device=dri
+  - --filesystem=home
+  - --filesystem=xdg-run/dconf
+  - --filesystem=~/.config/dconf:ro
+  - --env=DCONF_USER_CONFIG_DIR=.config/dconf
+  - --talk-name=ca.desrt.dconf
```
Comment (hfiguiere) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:20]: you don't need that. (lines 17-20)

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 15]
```diff
@@ -0,0 +1,50 @@
...
+rename-desktop-file: outlook-for-linux.desktop
+rename-icon: outlook-for-linux
+modules:
+  - name: outlook-for-linux
+    buildsystem: simple
+
+    sources:
+      - type: file
+        url: https://github.com/mahmoudbahaa/outlook-for-linux/releases/download/v1.3.13-outlook/outlook-for-linux_1.3.13_amd64.deb
```
Comment (hfiguiere) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:29]: does this include a vulneraable Electron runtime?

Also why no aarch64?

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 16]
```diff
@@ -0,0 +1,50 @@
...
+          - export TMPDIR="$XDG_RUNTIME_DIR/app/$FLATPAK_ID"
+          - exec zypak-wrapper /app/outlook-for-linux/outlook-for-linux "$@"
+
+    build-commands:
+      - ar x outlook-for-linux_*.deb
+      - rm -f outlook-for-linux_*.deb
+      - tar -xf data.tar.xz
+      - rm -f control.tar.gz data.tar.xz debian-binary
+      - cp -r opt/* usr/* $FLATPAK_DEST
+      - rm -rf opt usr
```
Comment (hfiguiere) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:46]:
```suggestion
```

File: io.github.mahmoudbahaa.outlook_for_linux.yml [hunk 17]
```diff
@@ -0,0 +1,50 @@
...
+        dest-filename: outlook-for-linux.sh
+        commands:
+          - export TMPDIR="$XDG_RUNTIME_DIR/app/$FLATPAK_ID"
+          - exec zypak-wrapper /app/outlook-for-linux/outlook-for-linux "$@"
+
+    build-commands:
+      - ar x outlook-for-linux_*.deb
+      - rm -f outlook-for-linux_*.deb
+      - tar -xf data.tar.xz
+      - rm -f control.tar.gz data.tar.xz debian-binary
```
Comment (hfiguiere) [inline io.github.mahmoudbahaa.outlook_for_linux.yml:44]:
```suggestion
```

File: .flatpak-builder/ccache/disabled/ccache.conf [hunk 18]
```diff
@@ -0,0 +1 @@
+disable = true
```
Comment (hfiguiere) [inline .flatpak-builder/ccache/disabled/ccache.conf:1]: why?

Comment (mahmoudbahaa) [reply .flatpak-builder/ccache/disabled/ccache.conf:1]: sorry got pushed by mistake

File: flathub.json (hunk 1 above)
Comment (mahmoudbahaa) [reply flathub.json:2]: will add it once stabilize x64 first

File: io.github.mahmoudbahaa.outlook_for_linux.appdata.xml [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+	<summary>Unofficial Microsoft Outlook client for Linux using Electron</summary>
+	<description>
+		<p>Unofficial Microsoft Outlook client for Linux using Electron.</p>
+		<p>It uses the Web App and wraps it as a standalone application using Electron.</p>
+		<p>A clone of <b>teams-for-linux</b> but for outlook. All credits for authors of that repo</p>
+	</description>
+	<url type="homepage">https://github.com/mahmoudbahaa/outlook-for-linux</url>
+	<url type="help">https://github.com/mahmoudbahaa/outlook-for-linux/blob/develop-outlook/KNOWN_ISSUES.md</url>
+	<url type="bugtracker">https://github.com/mahmoudbahaa/outlook-for-linux/issues</url>
+	<url type="details">https://github.com/IsmaelMartinez/teams-for-linux</url>
```
[the remark this answers is not in the document]
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.appdata.xml:16]: ok will fix

File: flathub.json (hunk 1 above)
Comment (hfiguiere) [reply flathub.json:2]: what is it to stabilize? What I just read is "it won't happen"

**Comment by mahmoudbahaa** [comment]
bot, build io.github.mahmoudbahaa.outlook_for_linux

File: io.github.mahmoudbahaa.outlook_for_linux.appdata.xml (hunk 19 above)
[the remark this answers is not in the document]
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.appdata.xml]: done

File: .flatpak-builder/ccache/disabled/ccache.conf (hunk 18 above)
Comment (mahmoudbahaa) [reply .flatpak-builder/ccache/disabled/ccache.conf]: done

File: io.github.mahmoudbahaa.outlook_for_linux.yml (hunk 16 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.yml]: done

File: io.github.mahmoudbahaa.outlook_for_linux.yml (hunk 15 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.yml:23]: no  electron 27 https://github.com/mahmoudbahaa/outlook-for-linux/blob/develop-outlook/package.json 
added aarch64

File: io.github.mahmoudbahaa.outlook_for_linux.yml (hunk 14 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.yml]: done

File: io.github.mahmoudbahaa.outlook_for_linux.yml (hunk 13 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.yml]: done

File: io.github.mahmoudbahaa.outlook_for_linux.yml (hunk 11 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.yml]: removed

**Comment by mahmoudbahaa** [comment]
Queued test build for io.github.mahmoudbahaa.outlook_for_linux.

File: io.github.mahmoudbahaa.outlook_for_linux.yml (hunk 15 above)
Comment (mahmoudbahaa) [reply io.github.mahmoudbahaa.outlook_for_linux.yml]: sorry it was 24.6.7 in the latest release i made a build from the pre-release with updated electron 27.0.0 and updated urls in manifest

Comment (hfiguiere) [reply io.github.mahmoudbahaa.outlook_for_linux.yml]: 24.6.7 is vulnerable.
